projects/jobs/jobs.groovy

```diff
@@ -12,24 +12,72 @@ def projectFolder = folder(projectFolderName)
 def cartridgeManagementFolderName= projectFolderName + "/Cartridge_Management"
 def cartridgeManagementFolder = folder(cartridgeManagementFolderName) { displayName('Cartridge Management') }
 
-// Cartridge List
-def cartridge_list = []
-readFileFromWorkspace("${WORKSPACE}/cartridges.txt").eachLine { line ->
-    cartridge_repo_name = line.tokenize("/").last()
-    local_cartridge_url = cartridgeBaseUrl + "/" + cartridge_repo_name
-    cartridge_list << local_cartridge_url
-}
-
-
 // Jobs
 def loadCartridgeJob = freeStyleJob(cartridgeManagementFolderName + "/Load_Cartridge")
 def loadCartridgeCollectionJob = workflowJob(cartridgeManagementFolderName + "/Load_Cartridge_Collection")
-
-
 // Setup Load_Cartridge
 loadCartridgeJob.with{
     parameters{
-        choiceParam('CARTRIDGE_CLONE_URL', cartridge_list, 'Cartridge URL to load')
+        extensibleChoiceParameterDefinition {
+          name('CARTRIDGE_CLONE_URL')
+          choiceListProvider {
+            systemGroovyChoiceListProvider {
+              scriptText('''
+              @Grab('org.yaml:snakeyaml:1.17')
+              import org.yaml.snakeyaml.Yaml
+              import org.yaml.snakeyaml.parser.ParserException
+              import jenkins.model.*
+
+              nodes = Jenkins.instance.globalNodeProperties
+              nodes.getAll(hudson.slaves.EnvironmentVariablesNodeProperty.class)
+              envVars = nodes[0].envVars
+
+              def URLS = envVars['CARTRIDGE_SOURCES'];
+
+              if (URLS == null) {
+                return ['[ERROR] CARTRIDGE_SOURCES Jenkins environment variable has not been set'];
+                println "[ERROR] CARTRIDGE_SOURCES Jenkins environment variable has not been set";
```

Comment:
@kramos It isn't anything to do with the browser session, it's the first time you try and build with parameters on a fresh ADOP Jenkins, where the population script has never been evaluated. I haven't personally tested it with this plugin, but it maybe an issue specific to the plugin I was using since I've only ever seen it happen with Dynamic Choice Parameter.